keanu-python/keanu/vertex/ops.py

```diff
@@ -0,0 +1,104 @@
+from typing import Any
+import keanu as kn
+
+class VertexOps:
+    """
+    __array_ufunc__ is a NumPy thing that enables you to intercept and handle the numpy operation.
+    Without this the right operators would fail.
+    See https://docs.scipy.org/doc/numpy-1.13.0/neps/ufunc-overrides.html
+    """
+    def __array_ufunc__(self, ufunc: Any, method: Any, *inputs: Any, **kwargs: Any) -> Any:
+        methods = {
+            "equal" : VertexOps.__eq__,
+            "not_equal" : VertexOps.__ne__,
+            "add" : VertexOps.__radd__,
+            "subtract" : VertexOps.__rsub__,
+            "multiply" : VertexOps.__rmul__,
+            "power" : VertexOps.__rpow__,
+            "true_divide" : VertexOps.__rtruediv__,
+            "floor_divide" : VertexOps.__rfloordiv__,
+            "greater" : VertexOps.__lt__,
+            "greater_equal" : VertexOps.__le__,
+            "less" : VertexOps.__gt__,
+            "less_equal" : VertexOps.__ge__,
+        }
+        if method == "__call__":
+            try:
+                dispatch_method = methods[ufunc.__name__]
+                return dispatch_method(inputs[1], inputs[0])
+            except KeyError:
+                raise NotImplementedError("NumPy ufunc of type %s not implemented" % ufunc.__name__)
+        else:
+            raise NotImplementedError("NumPy ufunc method %s not implemented" % method)
+
+    def __add__(self, other: Any) -> Any:
```

Comment:
We can be more specific than `Any` here can't we? We know the param will be `numpy array` or `vertex`?

Reply:
Due to circular dependencies between `VertexOps` and `Vertex`+`vartypes` I can't. But this just means we need to decouple some modules in the future.